expression/builtin_time.go

```diff
@@ -3434,6 +3434,14 @@ func (b *builtinAddDateStringIntSig) evalTime(row chunk.Row) (types.Time, bool,
 		return types.ZeroTime, true, err
 	}
 
+	dateStr, isNull, err := b.args[0].EvalString(b.ctx, row)
```

Comment:
Should this logic be in `b.getDateFromString`? 

Maybe we should fix them in all these signatures `builtin{Add|Sub}DateString{String|Int|Real|decimal}Sig`

Reply:
And we should also consider the vectorized version in `b.vecEvalTime` and better to add some comparison tests in https://github.com/pingcap/tidb/blob/baf6c99b956503b3122d8db463d7016c7b178fef/expression/builtin_time_vec_test.go#L243-L244